### PR TITLE
Implement If-Match value checking.

### PR DESCRIPTION
**Pull request by mikemoody-amido**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-825

## Describe this PR

Implemented checking of If-Match header value on person update.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

# Next steps

Do not merge approved PR until FE has implemented including the If-Match header value as without it the update person call will fail.

### CODE REVIEW

File: PersonApi/V1/Gateways/DynamoDbGateway.cs [hunk 1]
```diff
@@ -46,14 +52,22 @@ public async Task<Person> PostNewPersonAsync(CreatePersonRequestObject requestOb
         [LogCall]
         public async Task<UpdateEntityResult<PersonDbEntity>> UpdatePersonByIdAsync(UpdatePersonRequestObject requestObject, string requestBody, PersonQueryObject query)
         {
-            _logger.LogDebug($"Calling IDynamoDBContext.SaveAsync to update id {query.Id}");
-
             var existingPerson = await _dynamoDbContext.LoadAsync<PersonDbEntity>(query.Id).ConfigureAwait(false);
             if (existingPerson == null) return null;
 
+            // Check the If-Match header value
+            string ifMatch = _httpContextAccessor.HttpContext.Request.Headers.GetHeaderValue(HeaderConstants.IfMatch);
```
Comment (amidomichael) [inline PersonApi/V1/Gateways/DynamoDbGateway.cs:59]: I am hesitant to introduce a dependency on the http request in the gateway. I would think you'd pass the version number in, and get it from the header in the controller...

Comment (mikemoody-amido) [reply PersonApi/V1/Gateways/DynamoDbGateway.cs:59]: Fair point.

File: PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs [hunk 2]
```diff
@@ -39,6 +42,9 @@ public DynamoDbGatewayTests(AwsIntegrationTests<Startup> dbTestFixture)
             _dynamoDb = dbTestFixture.DynamoDbContext;
             _mockUpdater = new Mock<IEntityUpdater>();
             _logger = new Mock<ILogger<DynamoDbGateway>>();
+            _mockContextAccessor = new Mock<IHttpContextAccessor>();
```
Comment (amidomichael) [inline PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:45]: is this still needed?

Comment (mikemoody-amido) [reply PersonApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:45]: No. I thought I'd removed all of that. Will delete.

**Review by amidomichael — APPROVED** [review]

**Review by humulla — APPROVED** [review]

**Pull request merged**